orttraining/orttraining/test/python/perf_log/ort_module_perf_test_tools.py

```diff
@@ -0,0 +1,231 @@
+        perf_metrics_duration.microseconds / args.train_steps
+
+    perf_metrics['Throughput'] = \
+        args.batch_size * args.train_steps / perf_metrics_duration.seconds
+
+    perf_metrics['StabilizedThroughput'] = 0    # TODO
+    perf_metrics['EndToEndThroughput'] = 0      # TODO
+    perf_metrics['TotalTime'] = perf_metrics_duration.seconds
+
+    perf_metrics['AvgCPU'] = 0                  # TODO
+    perf_metrics['Memory'] = 0                  # TODO
+    perf_metrics['RunConfig'] = 'na'
+    perf_metrics['Time'] = completion_time.strftime("%Y-%m-%d %H:%M:%S")
+
+    log_perf_metrics(perf_metrics, mysql_server_name, power_bi_user_name, power_bi_password, power_bi_database,
+        ort_repo_path)
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--mysql_server_name', help='Perf dashboard mysql server name')
+    parser.add_argument('--power_bi_user_name', help='Power BI user name')
+    parser.add_argument('--password', help='password', default=None)
+    parser.add_argument('--database', help='The dashboard database')
+    return parser.parse_args()
+
+def ConnectAndInsertPerfMetrics(mysql_server_name, power_bi_user_name, password, database, perf_metrics):
+    conn = ConnectToPerfDashboardDb(mysql_server_name, power_bi_user_name, password, database)
+    insert_table_script_values = insert_table_script.format(
+        Model=perf_metrics['Model'],
+        BatchId=perf_metrics['BatchId'],
+        CommitId=perf_metrics['CommitId'],
+        ModelName=perf_metrics['ModelName'],
+        DisplayName=perf_metrics['DisplayName'],
+        UseMixedPrecision=perf_metrics['UseMixedPrecision'],
+        UseAutoCast=perf_metrics['UseAutoCast'],
+        UseDeepSpeed=perf_metrics['UseDeepSpeed'],
+        Optimizer=perf_metrics['Optimizer'],
+        BatchSize=perf_metrics['BatchSize'],
+        SeqLen=perf_metrics['SeqLen'],
+        PredictionsPerSeq=perf_metrics['PredictionsPerSeq'],
+        NumOfBatches=perf_metrics['NumOfBatches'],
+        WeightUpdateSteps=perf_metrics['WeightUpdateSteps'],
+        Round=perf_metrics['Round'],
+        GradAccSteps=perf_metrics['GradAccSteps'],
+        AvgTimePerBatch=perf_metrics['AvgTimePerBatch'],
+        Throughput=perf_metrics['Throughput'], 
+        StabilizedThroughput=perf_metrics['StabilizedThroughput'],
+        EndToEndThroughput=perf_metrics['EndToEndThroughput'],
+        TotalTime=perf_metrics['TotalTime'],
+        AvgCPU=perf_metrics['AvgCPU'],
+        Memory=perf_metrics['Memory'],
+        RunConfig=perf_metrics['RunConfig'],
+        Time=perf_metrics['Time'])
+    conn.cursor().execute(insert_table_script_values)
+    conn.commit()
+    conn.cursor().close()
+    conn.close()
+    print("Done.")
+
+if __name__ == '__main__':
+    args = parse_arguments()
+    conn = ConnectToPerfDashboardDb(args.mysql_server_name, args.power_bi_user_name, args.password, args.database)
+    conn.cursor().execute(create_table_script)
```

Comment:
>create_table_script [](start = 26, length = 19)

would this failed if table already exists?

Reply:
I put this code in so that we have the table definition in repo. otherwise if we lost the database, we lost it definition. You are correct that this is just a onetime thing.